### PR TITLE
Device.reinstall: support, including different OS / custom iPXE

### PR DESCRIPTION
**Pull request by grahamc**

### CODE REVIEW

File: packet/Device.py [hunk 1]
```diff
@@ -65,6 +65,18 @@ def update(self):
     def delete(self):
         return self.manager.call_api("devices/%s" % self.id, type="DELETE")
 
+    def reinstall(self, operating_system=None, ipxe_script_url=None):
+        params = {"type": "reinstall"}
+        if operating_system is not None:
+            params["operating_system"] = operating_system
+        if ipxe_script_url is not None:
```
Comment (mmlb) [inline packet/Device.py]: Was going to mention about erroring if `operating_system` is set to !`custom_ipxe` but `ipxe_script_url` is passed in but then noticed we do this override in the regular provision flow so nm.

**Comment by mmlb** [comment]
changelog please @grahamc

File: packet/Device.py (hunk 1 above)
Comment (grahamc) [reply packet/Device.py]: Yeah, the other thing is there is a valid UUID name for custom_ipxe too ... and then just "meh"'d my way to here :P

**Comment by grahamc** [comment]
Changelog'd.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [1.43.0] - 2020-02-24
+### Added
+- Support for reinstalling the operating system to a device, including
+  changing the operating system.
```
Comment (mmlb) [inline CHANGELOG.md]: just one line please (https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line)

Comment (grahamc) [reply CHANGELOG.md]: done!

**Pull request merged**